sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+    public static final String URL_LOGOUT                   = "logout";
+    public static final String URL_SENSOR_DEVICE            = "device";
+    public static final String URL_SENSORS                  = "sensors";
+    public static final String URL_USERS                    = "users";
+    public static final String URLE_UPLOAD_MULTIPLE_SENSORS = "sensors/data";
+    public static final String URL_DATA                     = "data";
+    public static final String URL_DEVICES                  = "devices";
+
+    public static final String URL_JSON_SUFFIX              = ".json";
+
+    public static final String HTTP_METHOD_POST = "POST";
+    public static final String HTTP_METHOD_GET = "GET";
+
+
+    /**
+     * Key for getting the http response code from the Map object that is returned by
+     * {@link nl.sense_os.datastorageengine.CommonSenseProxy#request(String, JSONObject, String, String)}
+     */
+    public static final String RESPONSE_CODE = "http response code";
+    /**
+     * Key for getting the response content from the Map object that is returned by
+     * {@link nl.sense_os.datastorageengine.CommonSenseProxy#request(String, JSONObject, String, String)}
+     */
+    public static final String RESPONSE_CONTENT = "content";
+
+    /**
+     * Default constructor for the DSECommonSenseProxy.
+     *
+     * Takes an app key that will be used throughout the proxies lifetime. Needs to know whether to talk to the live server or the staging server.
+     * This cannot be changed during the proxies lifetime. If you need to change this you simply init a new commonsense proxy.
+     *
+     * @param useLiveServer	If true, the live server will be used. If false, the staging server will be used.
+     * @param theAppKey		An application key that identifies the application to the commonsense server. Cannot be empty.
+     */
+    public CommonSenseProxy (boolean useLiveServer, String theAppKey)
+    {
+        APP_KEY = theAppKey;
+        if(useLiveServer) {
+            URL_BASE = BASE_URL_LIVE;
+            URL_AUTH = BASE_URL_AUTHENTICATION_LIVE;
+        } else {
+            URL_BASE = BASE_URL_STAGING;
+            URL_AUTH = BASE_URL_AUTHENTICATION_STAGING;
+        }
+    }
+
+
+    /**
+     * Login a user
+     *
+     * If a user is already logged in this call will also return the session ID.
+     *
+     * @param username		A user account in commonsense is uniquely identified by a username. Cannot be empty.
+     * @param password		A password in commonsense does not have any specific requirements.
+     *                      It will be MD5 hashed before sending to the server so the user does not have to provide a hashed password. Cannot be empty.
+     * @throws IOException and JSONException
+     * @return				Session ID. Will be null if the call fails.
+     */
+    public String loginUser( String username, String password) throws IOException, JSONException
```

Comment:
@excaliburnan would not use JSONException here, catch it internally and throw an appropriate exception